src/main/java/darwin/parser/InputParser.java

```diff
@@ -0,0 +1,92 @@
+package darwin.parser;
+
+import darwin.command.Command;
+import darwin.command.ExitCommand;
+import darwin.command.ListCommand;
+import darwin.command.MarkCommand;
+import darwin.command.UnmarkCommand;
+import darwin.command.DeleteCommand;
+import darwin.command.FindCommand;
+import darwin.command.AddTaskCommand;
+import darwin.exception.IllegalTaskArgumentException;
+import darwin.exception.IllegalTaskTypeException;
+import darwin.task.Task;
+
+public class InputParser {
+    private static final String EMPTY_ARGS = "";
+
+    /**
+     * Parses the input string into a Command object.
+     * @param in user input
+     * @return Command based on input
+     * @throws IllegalTaskTypeException if task type is invalid
+     * @throws IllegalTaskArgumentException if task arguments are invalid
+     */
+    public Command parse(String in) throws IllegalTaskTypeException, IllegalTaskArgumentException {
+        in = in.trim();
+        String[] parts = in.split("\\s+", 2);
+        String cmd = parts[0];
+        String args = parts.length == 2 ? parts[1] : InputParser.EMPTY_ARGS;
+
+        switch (cmd) {
+        case ExitCommand.CMD_WORD -> {
```

Comment:
Since your case bodies are only 1 line long, maybe you could use the default syntax given in the coding standard?
Cases would not use the arrow and braces but instead only have a colon.